library_api/controllers/book_controller.go

```diff
@@ -0,0 +1,75 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"library_api/entities"
+	"library_api/repositories"
+	"net/http"
+	"sort"
+	"strconv"
+)
+
+func CreateBooks(w http.ResponseWriter, r *http.Request) { // TODO discuss about create action
+	name := r.FormValue("name")
+	author := r.FormValue("author")
+	year, err := strconv.Atoi(r.FormValue("year"))
```

Comment:
for this, you can use https://calvinfeng.gitbook.io/gonotebook/idioms/custom-json-marshaling